tools/SetupFlow/DevHome.SetupFlow/Models/ConfigureTargetTask.cs

```diff
@@ -0,0 +1,435 @@
+
+                CreateCorrectiveActionPanel(ExtensionAdaptiveCardSession).GetAwaiter().GetResult();
+
+                IsAdaptiveCardPresentInUI = true;
+                AddMessage(_stringResource.GetLocalized(StringResourceKey.ConfigureTargetApplyConfigurationActionNeeded, UserNumberOfAttempts++, UserMaxNumberOfAttempts), MessageSeverityKind.Warning);
+                Log.Logger?.ReportInfo(Log.Component.ConfigurationTarget, $"adaptive card receieved from extension");
+            }
+            else
+            {
+                // Adaptive card session was not sent, so we check if there are any errors or due to applying a configuration unit/set.
+                var wrapper = new SDKConfigurationSetChangeWrapper(progressData, _stringResource);
+                if (wrapper.IsErrorMessagePresent)
+                {
+                    Log.Logger?.ReportError(Log.Component.ConfigurationTarget, $"Target experienced an error while applying the configuration: {wrapper.GetErrorMessageForLogging()}");
+                    AddMessage(wrapper.GetErrorMessagesForDisplay(), MessageSeverityKind.Error);
+                }
+
+                // In the future we need to add more messaging to the UI for the user to understand what is happening. It is on the extension to provide
+                // us with this messaging. Right now we only get error information and information about which configuration units are/were applied. However
+                // there is no way for us to know what the extension is doing, it may not have started configuration yet but may simply be installing prerequisites.
+            }
+        }
+        catch (Exception ex)
+        {
+            Log.Logger?.ReportError(Log.Component.ConfigurationTarget, $"Failed to process configuration progress data on target machine.'{ComputeSystemName}'", ex);
+        }
+    }
+
+    public void OnApplyConfigurationOperationCompleted(object sender, SDK.ApplyConfigurationResult applyConfigurationResult)
+    {
+        // apply configuration set result is used to check if the configuration set was applied successfully, while open configuration
+        // set result is used to check if WinGet was able to open the configuration file successfully.
+        var applyConfigSetResult = applyConfigurationResult.ApplyConfigurationSetResult;
+        var openConfigResult = applyConfigurationResult.OpenConfigurationSetResult;
+        var resultCode = applyConfigurationResult.ResultCode;
+        var resultInformation = applyConfigurationResult.ResultDescription;
+
+        try
+        {
+            Result = new SDKApplyConfigurationResult(
+                resultCode, resultInformation, new SDKApplyConfigurationSetResult(applyConfigSetResult), new SDKOpenConfigurationSetResult(openConfigResult, _stringResource));
+
+            // Check if there were errors while opening the configuration set.
+            if (!Result.OpenConfigSucceeded)
```

Comment:
I think we need to check resultCode (or Result.ResultCode) first. The operation could fail before getting to OpenConfigurationSetResult.

Reply:
I'll update these when we have the update sdk changes into main